src/Openplatform/OpenplatformOauthApiClient.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace DanskernesDigitaleBibliotek\AgencyAuthBundle\Openplatform;
+
+use DanskernesDigitaleBibliotek\AgencyAuthBundle\Exception\OpenPlatformException;
+use DanskernesDigitaleBibliotek\AgencyAuthBundle\Security\User;
+use DanskernesDigitaleBibliotek\AgencyAuthBundle\Utils\Logger;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class OpenplatformOauthApiClient
+{
+    /**
+     * OpenplatformOauthApiClient constructor.
+     *
+     * @param string $clientId
+     *   Open Platform id
+     * @param string $clientSecret
+     *   Open Platform secret
+     * @param string $endPoint
+     *   Open Platform introspection URL
+     * @param HttpClientInterface $client
+     *   Http client for calls to Open Platform
+     * @param Logger $logger
+     *   Bundle logger service
+     */
+    public function __construct(private readonly string $clientId, private readonly string $clientSecret, private readonly string $endPoint, private readonly HttpClientInterface $client, private readonly Logger $logger)
+    {
+    }
```

Comment:
I prefer this code style for constructor property promotion (but don't no what symfony do):
```
public function __construct(
        string $name, 
        string $email, 
        DateTimeImmutable $birth_date
    ) {
        $this->name = $name;
        $this->email = $email;
        $this->birth_date = $birth_date;
    }
```

Which is also how the original RFC defines it: https://wiki.php.net/rfc/constructor_promotion

Reply:
Actually this is how the RFC defines it:
```
class Point {
    public function __construct(
        public float $x = 0.0,
        public float $y = 0.0,
        public float $z = 0.0,
    ) {}
}
```

Reply:
Fixed. Except `{}`. PSR-12 doesn't allow them on the same line: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5632